lib/plugins/game.js

```diff
@@ -25,7 +30,12 @@ function inject (bot, options) {
   function handleRespawnPacketData (packet) {
     bot.game.levelType = packet.levelType ?? (packet.isFlat ? 'flat' : 'default')
     bot.game.hardcore = packet.isHardcore ?? Boolean(packet.gameMode & 0b100)
-    bot.game.gameMode = packet.gamemode || parseGameMode(packet.gameMode)
+    // Either a respawn packet or a login packet. Depending on the packet it can be "gamemode" or "gameMode"
+    if (bot.supportFeature('spawnRespawnWorldDataField')) { // 1.20.5
+      bot.game.gameMode = packet.gamemode
+    } else {
+      bot.game.gameMode = parseGameMode(packet.gamemode ?? packet.gameMode)
+    }
```

Comment:
I don't understand how this is different from the current code. It does the same thing.

Reply:
Pre 1.20.5, the respawn packet and the login packet used numbers to indicate what gamemode the player is in. The problem is that with one packet, it's `gamemode`, and with the other, it's `gameMode`.

Reply:
`packet.gamemode` could be a number. But game.gameMode needs to be a string. Also,`||` is wrong as it will be falsy if the gamemode is 0. And 0 is a valid gamemode in some versions where the packet has gamemode as a number.

Reply:
This line was broken before `bot.game.gameMode = packet.gamemode || parseGameMode(packet.gameMode)`. It only works correctly in versions after 1.20.5. In versions pre 1.20.5, it would set game.gameMode to a number for gamemodes other than survival. No one noticed that because most people use survival mode and the test was not working, and the change was only made recently.

Reply:
`gamemode` lower case is only defined on 1.20.5+ because this function is called with a SpawnInfo object. `gameMode` is the previous number field. `||` can be replaced with `??`, but otherwise for the else condition it should be this:

```js
      bot.game.gameMode = parseGameMode(packet.gameMode)
```

`parseGameMode` can likely just be set to `gameModes[gameModeBits] || 'survival']`

I think it may be worthwhile to just backport this change inside minecraft-data to 1.20.5 to all the old versions so we can get rid of the logic here

Reply:
> `gamemode` lower case is only defined on 1.20.5+ because this function is called with a SpawnInfo object.

No, the two different cases have been used before. Handle respawn is used by the respawn packet and the login packet. They both use different cases for the same field pre 1.20.5.

See https://prismarinejs.github.io/minecraft-data/?v=1.12.2&d=protocol#toClient_respawn

And

https://prismarinejs.github.io/minecraft-data/?v=1.12.2&d=protocol#toClient_login

The same protocol uses different versions of the field gamemode.

Reply:
I think it's better to focus on the tests instead of the implementation. The protocol is inconsistent and changes a lot across versions. Trying to figure out what the implementation should look like by looking really hard at the code always fails in mineflayer. That's why there are tests. They tell you if something works or not. The current implementation does not work for pre 1.20.5. Your changes to game.js to update past 1.20.5 only when through review because the gamemode test was not working and did not catch the bug with the different cased gamemode fields.

Reply:
Yes what I mean is there are things we can fix in minecraft-data to avoid the need to do special handling in mineflayer. 

Can you reopen the PR against the 1.21.2 branch?